### PR TITLE
[FIX] use discovery api instead of reading CRD

### PR DESCRIPTION
**Pull request by skrishnan-sap**
CHANGES:
- use discovery API to check existence of `ServiceMonitor` capability rather than fetching the `CustomResourceDefinition`
- remove unused `apiextensions` typed client
- adjust tests - `common_test` enhanced to supply discoverable resources
- refactor `ServiceMonitor` reconciliation to reduce complexity (sonar) 

Controller image for testing: `ghcr.io/skrishnan-sap/cap-operator/controller:sha-444fa21`

### CODE REVIEW

**Comment by Pavan-SAP** [comment]
Looks good.. can you also push the chart update to remove crd read rbac to test this completely:
- apiGroups: ["apiextensions.k8s.io"]
  resources: ["customresourcedefinitions"]
  verbs: ["get"]

**Comment by skrishnan-sap** [comment]
> Looks good.. can you also push the chart update to remove crd read rbac to test this completely:
> 
> * apiGroups: ["apiextensions.k8s.io"]
>   resources: ["customresourcedefinitions"]
>   verbs: ["get"]

Yes, I am waiting for this to be merged and released before merging https://github.com/SAP/cap-operator-lifecycle/pull/91.

There is built-in `ClusterRoleBinding` named `system:discovery` which grants access to the discovery API to all subjects which are part of the group `system:authenticated` (which is every authenticated access).

File: internal/controller/reconcile-capapplicationversion.go
Comment (Pavan-SAP) [inline internal/controller/reconcile-capapplicationversion.go:452]: use:
`monitoringGroupVersion` instead of "monitoring.coreos.com/v1"

File: internal/controller/common_test.go
Comment (Pavan-SAP) [inline internal/controller/common_test.go:48]: we could also remove apiExtScheme and its usage right?

File: internal/controller/common_test.go
Comment (Pavan-SAP) [inline internal/controller/common_test.go:239]: remove usage of apiExtScheme here

**Review by Pavan-SAP — COMMENTED** [review]
minor issues inline

**Comment by Pavan-SAP** [comment]
> Yes, I am waiting for this to be merged and released before merging [SAP/cap-operator-lifecycle#91](https://github.com/SAP/cap-operator-lifecycle/pull/91).
> 
> There is built-in `ClusterRoleBinding` named `system:discovery` which grants access to the discovery API to all subjects which are part of the group `system:authenticated` (which is every authenticated access).

Yes exactly, I meant to test by removing the CRD read RBAC which I can now do with the helm chart from your branch. But once the minor changes (previous comment) are done - we can just merge this.

**Comment by skrishnan-sap** [comment]
updated test image - `ghcr.io/skrishnan-sap/cap-operator/controller:sha-444fa21`

**Review by Pavan-SAP — APPROVED** [review]
✅

**Pull request merged**
